File: word_counter_js/spec/word_counter_integration_spec.js
var request = require('supertest');

var app = require('../app');

describe("requests to the root path", function(){
  it('returns a 200 status code', function(done){
    request(app)
      .get('/')
      .expect(200, done);
  });

  it('returns a HTML format', function(done){
    request(app)
      .get('/')
      .expect('Content-Type', /html/, done);
  });

  it('returns an index file with Word Counter', function(done){
    request(app)
      .get('/')
      .expect(/word counter/i, done);
  });

});

describe('requests to the word_count path', function(){
  it('takes a user inputed string and test_word and returns number of instances of test_word in user enter string', function(done){
    request(app)
      .post('/word_count')
      .send({'string': 'hi hi hi', 'test_word': 'hi'})
      .expect(/3/, done);
  });
});

File: spec/word_counter_integration_spec.rb
require 'capybara/rspec'
require './app'
Capybara.app = Sinatra::Application
set(:show_exceptions, false)

describe 'word_counter path', { type: :feature } do
  it 'takes a user inputed string and test_word and returns number of instances of test_word in user enter string' do
    visit '/'
    fill_in 'string', with: "If Peter Piper picked a peck of pickled peppers, where's the peck of pickled peppers Peter Piper picked?"
    fill_in 'test_word', with: 'peck'
    click_button 'Send'
    expect(page).to have_content 2
  end
end
